src/hooks/index.ts

```diff
@@ -237,14 +237,9 @@ export const useTokenInfo = (
   return query.data
 }
 
-export const useIsTokenKnownNft = ({wallet, tokenId}: {wallet: YoroiWallet; tokenId: string}) => {
-  const tokenInfo = useTokenInfo({wallet, tokenId})
+export const useIsTokenKnownNft = ({wallet, fingerprint}: {wallet: YoroiWallet; fingerprint: string}) => {
   const {nfts} = useNfts(wallet)
-  return nfts.some(
-    (nft) =>
-      nft.metadata.policyId === tokenInfo.metadata.policyId &&
-      nft.metadata.assetNameHex === tokenInfo.metadata.assetName,
-  )
+  return nfts.some((nft) => nft.id === fingerprint)
```

Comment:
why is nft.id the asset fingerprint and not policyId.assetNameHex?

Reply:
yeah, let's keep tokens and nfts as similar as possible
sorry, switch back to policyId.assetNameHex
and add the asset fingerprint as a property